appservice/src/remoteDebug/startRemoteDebug.ts

```diff
@@ -83,44 +84,92 @@ async function startRemoteDebugInternal(context: IActionContext, site: ParsedSit
-            config.remoteRoot = '/home/site/wwwroot';
+async function getDebugPath() : Promise<string> {
+        // Try to map workspace folder source files to the remote instance
+        if (vscode.workspace.workspaceFolders) {
+            if (vscode.workspace.workspaceFolders.length === 1) {
+                return vscode.workspace.workspaceFolders[0].uri.fsPath;
+            } else {
+                // In this case we don't know which folder to use. Show a warning and proceed.
+                // In the future we should allow users to choose a workspace folder to map sources from.
+                var root = await selectMultirootWorkspaceTarget();
+                if (root)
+                    return root.fsPath;
+                else
+                    throw new Error(localize('remoteDebugNoFolders', 'Please select a workspace folder before attaching a debugger.'));
+            }
         } else {
-            // In this case we don't know which folder to use. Show a warning and proceed.
-            // In the future we should allow users to choose a workspace folder to map sources from.
-            void context.ui.showWarningMessage(localize('remoteDebugMultipleFolders', 'Unable to bind breakpoints from workspace when multiple folders are open. Use "loaded scripts" instead.'));
+            // vscode will throw an error if you try to start debugging without any workspace folder open
+            throw new Error(localize('remoteDebugNoFolders', 'Please open a workspace folder before attaching a debugger.'));
         }
-    } else {
-        // vscode will throw an error if you try to start debugging without any workspace folder open
-        throw new Error(localize('remoteDebugNoFolders', 'Please open a workspace folder before attaching a debugger.'));
     }
 
+async function getNodeDebugConfiguration(sessionId: string, portNumber: number, host: string): Promise<vscode.DebugConfiguration> {
+    const config: vscode.DebugConfiguration = {
+        name: sessionId,
+        type: 'node',
+        protocol: 'inspector',
+        remoteRoot: '/home/site/wwwroot',
+        request: 'attach',
+        address: host,
+        port: portNumber,
+    }
+    config.localRoot = await getDebugPath();
     return config;
 }
+
+async function getPythonDebugConfiguration(sessionId: string, portNumber: number, host: string): Promise<vscode.DebugConfiguration> {
+    const localRoot = await getDebugPath();
+    const config: vscode.DebugConfiguration = {
+        name: sessionId,
+        type: 'python',
+        request: 'attach',
+        connect: {
+            host: host,
+            port: portNumber,
+        },
+        pathMappings: [
+            {
+                localRoot: localRoot,
+                remoteRoot: '.',
+            },
+        ],
+    };
+
+    return config;
+}
+
+async function getDebugConfiguration(language: RemoteDebugLanguage): Promise<vscode.DebugConfiguration> {

```

Comment:
Please move this back to the top and try to minimize refactoring stuff which is making the diff more difficult to read than it needs to be

Reply:
I forget that declaration order doesn't matter. Fixed